NameGame/NameGame/Services/ImageService.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  ImageService.swift
+//  NameGame
+//
+//  Created by Tammy Le on 6/14/17.
+//  Copyright © 2017 WillowTree Apps. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+final class ImageService {
+    
+    let sharedCache = NSCache<NSString, UIImage>()
+    
+    private(set) var task: URLSessionDataTask? = nil
+    
+    
+    func fetchImage(userRecord: UserRecord, completion: @escaping (Result<UIImage>) -> Void) {
+        if let cacheImage = sharedCache.object(forKey: userRecord.headshot.url as NSString) {
```

Comment:
Same here, let's pull the caching responsibility from the service and build an `ImageCache` class to handle this responsibility.